assets/source/setup-guide/app/components/StatusLabel/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { Icon } from '@wordpress/components';
+import { Spinner } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { LABEL_STATUS as STATUS } from '../../constants';
+import './style.scss';
+
+const compClassName = 'pfw-status-label';
+
+const statusStyleName = {
+	[ STATUS.PENDING ]: `${ compClassName }--status-pending`,
+	[ STATUS.SUCCESS ]: `${ compClassName }--status-success`,
+};
+
+function getStatusPrefix( status ) {
+	switch ( status ) {
+		case STATUS.PENDING:
+			return <Spinner />;
+		case STATUS.SUCCESS:
+			return <Icon icon="yes-alt" />;
+	}
+}
+
+/**
+ * Renders a text with a prefix graphic marking.
+ *
+ * @param {Object} props React props.
+ * @param {import('../../constants').LABEL_STATUS} props.status Label status.
+ * @param {string} props.text Label text.
+ */
+export default function StatusLabel( { status, text } ) {
```

Comment:
> I'm thinking it would be better to make this `StatusLabel` more of a layout component and doesn't care about status, e.g. a `IconLabel` that accepts `icon` and `text`.

It's not only a layout component. There's [high dependent styling](https://github.com/woocommerce/pinterest-for-woocommerce/blob/70205b028a0c460753f32ea58057ef5e3dbe1b75/assets/source/setup-guide/app/components/StatusLabel/style.scss) that depends on the status. With the proposed `<IconLabel>` which only takes care of layout, then those status-related styles will be placed elsewhere.  Like the Overview of Catalog tab, these future reusable styles are [scattered around outside](https://github.com/woocommerce/pinterest-for-woocommerce/blob/70205b028a0c460753f32ea58057ef5e3dbe1b75/assets/source/setup-guide/app/style.scss#L618-L643).

![image](https://user-images.githubusercontent.com/17420811/144970324-e8538e17-47a7-4955-9a36-c6ef9cc65a01.png)

Rather than combining status icons and layout components, and setting the status icon and additional layout styles for the combined React elements in some external places, I think it is better to put highly dependent and reusable styles into a dedicated component.

The `<StatusLabel>` component and its `status` prop are similar in concept to `<Notice>` of `wp-components`, and even the *status icon + text* usages in **Overview** above could be migrated to use `<StatusLabel>` in the future.

Reply:
> There's [high dependent styling](https://github.com/woocommerce/pinterest-for-woocommerce/blob/70205b028a0c460753f32ea58057ef5e3dbe1b75/assets/source/setup-guide/app/components/StatusLabel/style.scss) that depends on the status. 

Yeah I noticed the styling. 🙂 

> With the proposed `<IconLabel>` which only takes care of layout, then those status-related styles will be placed elsewhere.

Yes, it is related to https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763011970, my thinking is that those status-related style would be defined in the parent level `VerifyButton`, the parent would dictate what to render, including the styling.

> Like the Overview of Catalog tab, these future reusable styles are [scattered around outside](https://github.com/woocommerce/pinterest-for-woocommerce/blob/70205b028a0c460753f32ea58057ef5e3dbe1b75/assets/source/setup-guide/app/style.scss#L618-L643).

I didn't see them as "scattered around outside" though. I think it is fine that the parent decide what to render and how it is styled. 

Here's an example of my thinking: Let's say that we have a defined yellow warning color, like the screenshot you provided. There _might_ be places where we want to have a different warning color in different pages, and in those cases we probably gonna need a way (React props, or CSS) for the page or parent component to override the defined default yellow warning color anyway. The defined yellow warning color serves as a good convenient consistent default; the parent should have a way to dictate and override the default. 

About "future reusable styles", yeah that is a good point. We can certainly refactor those styles and make them more reusable. 😄 👍